Fix #289: Notebook 5.0 compatibility

parente opened the pull request:
* Switch to conda dev setup
* Fix quick-setup process for nb 5.0
* Fix selected cell CSS
* Fix bokeh demo for 0.12.4
* Cleanup documentation for 5.0

parente commented:
@dtaeib, @jhpedemonte, @dalogsdon care to give this a look before I pull the trigger? It's all dev setup / install cleanup.

jhpedemonte commented on requirements-demo.txt:
```diff
@@ -0,0 +1,6 @@
+matplotlib
+pandas
+seaborn
+widgetsnbextension
```
Should any other reqs (in this file and others) also be version limited?

parente replied:
If they break on newer versions, better to get issues about them.  As it stood before all these updates, the versions used were quite old from a pinned docker image. Great for a dev env, but it gives no indication of what the real install experience is going to be for users.

jhpedemonte commented on Dockerfile.dev:
```diff
@@ -1,19 +0,0 @@
-
-# Setup a sphinx environment for doc building
-COPY docs/environment.yml /tmp/environment.yml
-RUN conda env create -f /tmp/environment.yml
```
So getting rid of Docker in favor of Conda: just easier/cleaner? Or is there another reason?

parente replied:
Consistency with other Python projects mainly.

jhpedemonte commented:
Any idea what travis failure is about (for last commit)?

parente commented:
I can't get the travis log to load at the moment. Seems have gone haywire since the S3 outage yesterday.

jhpedemonte commented:
I get same error as in travis. When running `make env`, I get this:

```
source activate dashboards && \
		pip install -r requirements-doc.txt && \
		npm install && \
		npm run bower && \
		pip install -e . && \
		jupyter dashboards quick-setup --sys-prefix &&
/bin/bash: -c: line 1: syntax error: unexpected end of file
```

jhpedemonte commented:
Working for the most part, except it can't find declarative widgets. Should that be added to one of the requirements files?

parente commented:
Added decl widgets but it's currently busted on install with matplotlib 2.0. Opened an issue about it on the widget repo.

I think I addressed all the issues noted here. Going to merge this in prep for 5.0.
